### PR TITLE
Feature/ios headless support

### PR DESCRIPTION
**Pull request by syrusakbary**
# Description

This PR is a revamp of #2516 - @NathHorrigan so we can bypass bors issues

# Review

- [ ] Add a short description of the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
@NathHorrigan should we delete the `xcuserdatad` files?

Note the `nathanhorrigan.xcuserdatad`
```
tests/integration/ios/DylibExample/DylibExample.xcodeproj/project.xcworkspace/xcuserdata/nathanhorrigan.xcuserdatad/UserInterfaceState.xcuserstate
```

**Comment by NathHorrigan** [comment]
> @NathHorrigan should we delete the `xcuserdatad` files?
> 
> Note the `nathanhorrigan.xcuserdatad`
> 
> ```
> tests/integration/ios/DylibExample/DylibExample.xcodeproj/project.xcworkspace/xcuserdata/nathanhorrigan.xcuserdatad/UserInterfaceState.xcuserstate
> ```

Yep! I've added a dedicated iOS `.gitignore`

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

File: tests/integration/ios/DylibExample/DylibExample.xcodeproj/xcuserdata/nathanhorrigan.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (syrusakbary) [inline tests/integration/ios/DylibExample/DylibExample.xcodeproj/xcuserdata/nathanhorrigan.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: We probably don't want to commit this

Comment (NathHorrigan) [reply tests/integration/ios/DylibExample/DylibExample.xcodeproj/xcuserdata/nathanhorrigan.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: I swear I removed this... sigh

**Comment by NathHorrigan** [comment]
bors try

**Comment by syrusakbary** [comment]
Why tests are failing seems unrelated to this PR. Unless the `wasmer_is_compiler_available` in the c-api is affecting it, which I think is not.

@Hywan can you take a look?

**Comment by NathHorrigan** [comment]
bors try

**Comment by NathHorrigan** [comment]
bors try

**Comment by Hywan** [comment]
I've started to review this PR. Thank you @NathHorrigan for the work! I'll merge it as soon as I've finished my review if everything is fine :-).

**Comment by NathHorrigan** [comment]
Thanks @Hywan  - I look forward to it!

**Comment by syrusakbary** [comment]
Apologizes for taking a bit longer than expected on review.

I did a deep review of the PR and everything looks great, I'll merge manually now (since I'm the one I created the PR I can't use bors for merging).

**Pull request merged**